pandas/core/generic.py

```diff
@@ -8577,16 +8578,16 @@ def _where(
         fill_value = bool(inplace)
         cond = cond.fillna(fill_value)
 
-        msg = "Boolean array expected for the condition, not {dtype}"
+        msg = "Boolean array expected for the condition, not "
```

Comment:
This is a legit use case for format in our code base - essentially you are binding the message to a variable some point later in the function